quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/storage/infinispan/CacheManagerFactory.java

```diff
@@ -361,6 +377,28 @@ private static void configureTransportStack(ConfigurationBuilderHolder builder)
         var transportConfig = builder.getGlobalConfigurationBuilder().transport();
         if (transportStack != null && !transportStack.isBlank()) {
             transportConfig.defaultTransport().stack(transportStack);
+        } else {
+            var tableName = "JGROUPSPING";
+            var attributes = Map.of(
+                  // Leave initialize_sql blank as table is already created by Keycloak
+                  "initialize_sql", "",
+                  // Explicitly specify clear and select_all SQL to ensure "cluster_name" column is used, as the default
+                  // "cluster" cannot be used with Oracle DB as it's a reserved word.
+                  "clear_sql", String.format("DELETE from %s WHERE cluster_name=?", tableName),
+                  "delete_single_sql", String.format("DELETE from %s WHERE address=?", tableName),
+                  "insert_single_sql", String.format("INSERT INTO %s values (?, ?, ?, ?, ?)", tableName),
+                  "select_all_pingdata_sql", String.format("SELECT address, name, ip, coord FROM %s WHERE cluster_name=?", tableName),
+                  "remove_all_data_on_view_change", "true",
+                  "stack.combine", "REPLACE",
+                  "stack.position", "PING"
+            );
+            var stackName = "keycloak-jdbc-ping";
+            var stack = List.of(new ProtocolConfiguration(JDBC_PING2.class.getSimpleName(), attributes));
+            builder.addJGroupsStack(new EmbeddedJGroupsChannelConfigurator(stackName, stack, null), "udp");
```

Comment:
Why not add the stack to the cache-ispn.xml? UDP is most likely blocked in some cloud environments, shouldn't be based on TCP?

Reply:
> Why not add the stack to the cache-ispn.xml?

Good point. Originally I thought that we were going to have to provide per vendor SQL, but now that's not the case I can define this declaratively.



> UDP is most likely blocked in some cloud environments, shouldn't be based on TCP?

The Keycloak default is `UDP`, which is why I have used that here. The Operator explicitly configures the transport stick as it relies on DNS_PING, so JGROUPSPING will never be used in that environment. Maybe there's an argument for changing the default from UDP to TCP, but I think that needs to be a separate issue.

Reply:
Or add 2 stacks, one for UDP and TCP. 
Are we changing the default to use JDBC_PING or keeping plain UDP?

Reply:
The aim of the JDBC_PING was to have something that works out-of-the-box in all places. As we use TCP in Kubernetes, and have tested it a lot, I'd +1 to move to TCP as well to have one thing less users need to worry about.

Reply:
I've created https://github.com/keycloak/keycloak/issues/33651 to change the default to TCP.

Reply:
> > Why not add the stack to the cache-ispn.xml?
> 
> Good point. Originally I thought that we were going to have to provide per vendor SQL, but now that's not the case I can define this declaratively.

A downside to defining the stack in `cache-ispn.xml` is that it's then not possible to use JDBC_PING2 with a custom `--cache-config-file`. For example they may want to use the default stacks, but provide explicit configuration of the caches. The current approach allows them to do this, but they would have to also provide `cache-stack` if they wanted to deviate from JDBC_PING2.

Reply:
Fair enough. I'm not going to enter into the "what if" spiral :rofl: 
It needs documentation that says that the stack is automatically available.